fn/memory.go

```diff
@@ -36,3 +36,10 @@ func CopySlice[T any](slice []T) []T {
 	copy(newSlice, slice)
 	return newSlice
 }
+
+// CopyBytes returns a deep copy of the given byte slice.
+func CopyBytes(b []byte) []byte {
```

Comment:
Isn't htis the same as `CopySlice[[]byte]`?

Reply:
https://go.dev/play/p/_D5C9txX2TI

Reply:
Yup, I misinterpreted the 'shallow copy' warning on `CopySlice`.